src/components/playback/playbackmanager.js

```diff
@@ -1531,9 +1587,51 @@ class PlaybackManager {
 
             player.setSubtitleStreamIndex(selectedTrackElementIndex);
 
+            // Also disable secondary subtitles when disabling the primary subtitles
+            if (selectedTrackElementIndex === -1) {
+                self.setSecondarySubtitleStreamIndex(selectedTrackElementIndex);
+            }
+
             getPlayerData(player).subtitleStreamIndex = index;
         };
 
+        self.setSecondarySubtitleStreamIndex = function (index, player) {
+            player = player || self._currentPlayer;
+            if (!self.playerHasSecondarySubtitleSupport(player)) return;
+            if (player && !enableLocalPlaylistManagement(player)) {
+                try {
+                    return player.setSecondarySubtitleStreamIndex(index);
+                } catch (e) {
+                    console.error(`AutoSet - Failed to set secondary track: ${e}`);
+                }
+            }
+
+            const currentStream = getCurrentSecondarySubtitleStream(player);
+
+            const newStream = getSubtitleStream(player, index);
+
+            if (!currentStream && !newStream) {
+                return;
+            }
+
+            const clearingStream = currentStream && !newStream;
+            const changingStream = currentStream && newStream;
+            const addingStream = !currentStream && newStream;
+            // Secondary subtitles are currently only handled client side
+            // Changes to the server code are required before we can handle other delivery methods
+            if (!clearingStream && (changingStream || addingStream) && getDeliveryMethod(newStream) !== 'External') {
```

Comment:
You're absolutely right. This can just be simplified to `newStream` I wrote all the cases before I knew exactly what the limitations would be for client-side only work. I intended to go back and update them as I understood under what conditions we could display a secondary track.
It turns out that it is only for new streams for the scope of this PR!
I'll simplify and clean this up!